src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/DeliveryRequestResolver.java

```diff
@@ -48,45 +47,49 @@ public boolean canResolve(
 
         final Colony colony = (Colony) manager.getColony();
         final CitizenData freeDeliveryMan = colony.getCitizenManager().getCitizens()
-                                        .stream()
-                                        .filter(c -> c.getCitizenEntity() != null
-                                                && requestToCheck.getRequest().getTarget().isReachableFromLocation(c.getCitizenEntity().getLocation()))
-                                        .filter(c -> c.getJob() instanceof JobDeliveryman)
-                                        .findFirst()
-                                        .orElse(null);
-
-        if (freeDeliveryMan == null)
-        {
-            return false;
-        }
-
-        return true;
+                                              .stream()
+                                              .filter(citizenData -> citizenData.getCitizenEntity()
+                                                                       .map(entityCitizen -> requestToCheck.getRequest()
+                                                                                               .getTarget()
+                                                                                               .isReachableFromLocation(entityCitizen.getLocation()))
+                                                                       .orElse(false))
+                                              .filter(c -> c.getJob() instanceof JobDeliveryman)
+                                              .findFirst()
+                                              .orElse(null);
+
+        return freeDeliveryMan != null;
     }
 
     @Nullable
     @Override
     public List<IToken<?>> attemptResolve(
-                                        @NotNull final IRequestManager manager, @NotNull final IRequest<? extends Delivery> request)
+      @NotNull final IRequestManager manager, @NotNull final IRequest<? extends Delivery> request)
     {
         if (manager.getColony().getWorld().isRemote)
         {
             return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1168)